src/concepts/essentials/binary.md

```diff
@@ -0,0 +1,74 @@
+# Dealing with Binary Data
+
+Caido plugins are written in JavaScript. When creating a plugin that must handle raw bytes, there are some caveats to be aware of as the backend language used by Caido is Rust.
+
+## UTF-8 Encoding
+
+For Unicode code points outside of the ASCII range <span style="color: #EBEBF599; font-style: italic">(U+0000 to U+007F)</span> multiple bytes are used.
+
+When encoding code points into multibyte UTF-8 characters, the bytes are prefixed with specific bit patterns. The first byte prefix indicates how many bytes are used, and any subsequent bytes begin with 10.
+
+<img alt="UTF-8 encoding chart." src="/_images/utf_chart.png" center/>
+
+This can be an issue if you want to create a Caido plugin that sends binary to a target server to see how it is processed.
+
+## Example
+
+Consider this `/packages/backend/src/index.ts` file:
+
+``` ts
+import { RequestSpecRaw } from "caido:utils";
+import { SDK, DefineAPI } from "caido:plugin";
+
+async function testSendRequest(sdk: SDK): Promise<void> {
+  console.log("Testing send request");
+  const req = new RequestSpecRaw("http://localhost:5555");
+  req.setRaw("GET /admin\x85 HTTP/1.1\r\nHost: HOST\r\n\r\n");
+
+  const res = await sdk.requests.send(req);
+  console.log(res?.response.getRaw().toText());
+}
+```
+
+The intention is to send the byte `\x85` with a binary value of `[1000 0101]`. However, JavaScript is interpreting it as the Unicode code point.
```

Comment:
code point `C2 85`.

Reply:
I thought the goal was to send just the byte 85?